src/main/resources/avro/sequenceAnnotationmethods.avdl

```diff
@@ -0,0 +1,165 @@
+    /**
+    The `Dataset` to search.
+    */
+    string datasetId;
+
+    /**
+    Specifies the maximum number of results to return in a single page.
+    If unspecified, a system default will be used.
+    */
+    union { null, int } pageSize = null;
+
+    /**
+    The continuation token, which is used to page through large result sets.
+    To get the next page of results, set this parameter to the value of
+    `nextPageToken` from the previous response.
+    */
+    union { null, string } pageToken = null;
+  }
+
+  /** This is the response from `POST /featuresets/search` expressed as JSON. */
+  record SearchFeatureSetsResponse {
+    /** The list of matching feature sets. */
+    array<org.ga4gh.models.FeatureSet> featureSets = [];
+
+    /**
+    The continuation token, which is used to page through large result sets.
+    Provide this value in a subsequent request to return the next page of
+    results. This field will be empty if there aren't any additional results.
+    */
+    union { null, string } nextPageToken = null;
+  }
+
+  /**
+  Gets a list of `FeatureSet` matching the search criteria.
+
+  `POST /featuresets/search` must accept a JSON version of
+  `SearchFeatureSetsRequest` as the post body and will return a JSON version
+  of `SearchFeatureSetsResponse`.
+  */
+  SearchFeatureSetsResponse searchFeatureSets(
+    /** This request maps to the body of `POST /featuresets/search` as JSON. */
+    SearchFeatureSetsRequest request) throws GAException;
+
+  /**************** /featuresets/{id} *******************/
+  /**
+  Gets a `FeatureSet` by ID.
+  `GET /featuresets/{id}` will return a JSON version of `FeatureSet`.
+  */
+  org.ga4gh.models.FeatureSet getFeatureSet(
+    /**
+    The ID of the `FeatureSet`.
+    */
+    string id) throws GAException;
+
+
+  /****************** /features/search *****************/
+  /**
+  This request maps to the body of `POST /features/search` as JSON.
+  */
+  record SearchFeaturesRequest {
+    /**
+    The annotation set to search within. Either `featureSetId` or
+    `parentId` must be non-empty.
+    */
+    union { null, string } featureSetId;
+
+    /**
+    Restricts the search to direct children of the given parent `feature`
+    ID. Either `featureSetId` or `parentId` must be non-empty.
+    */
+    union { null, string } parentId;
+
+    /**
+    Only return features on the reference with this ID.
+    */
+    union { null, string } referenceId = null;
+
+    /**
+    Required. The beginning of the window (0-based, inclusive) for which
+    overlapping features should be returned.  Genomic positions are
+    non-negative integers less than reference length.  Requests spanning the
+    join of circular genomes are represented as two requests one on each side
+    of the join (position 0).
+    */
+    long start;
+
+    /**
+    Required. The end of the window (0-based, exclusive) for which overlapping
+    features should be returned.
+    */
+    long end;
```

Comment:
Good catch, @sarahhunt! Furthermore, with our eventual transition to protobuf, we won't have the luxury of nullable fields, which makes this a no-brainer to remove the union with null here.